consumer/consumer.go

```diff
@@ -105,14 +108,15 @@ func newConsumer(
 	opts Options,
 	m metrics,
 ) *consumer {
-	r := bufio.NewReaderSize(conn, opts.ConnectionReadBufferSize())
-	w := bufio.NewWriterSize(conn, opts.ConnectionWriteBufferSize())
-	rw := bufio.NewReadWriter(r, w)
 	return &consumer{
-		opts:   opts,
-		mPool:  mPool,
-		encdec: proto.NewEncodeDecoder(rw, opts.EncodeDecoderOptions()),
-		rw:     rw,
+		opts:    opts,
+		mPool:   mPool,
+		encoder: proto.NewEncoder(opts.EncodeDecoderOptions().EncoderOptions()),
```

Comment:
I do plan to remove that and more stuff related to `EncodeDecoder` because the `EncodeDecoder` is pretty much only used in test now, but I'm going to do it in a different diff since that would require changing how I configure the producer and consumer, so I don't want to tangle that with this logic change.